### PR TITLE
[TR] PIM-4493: add empty value provider

### PR DESCRIPTION
**Pull request by sumbobyboys**
| Q                    | A
| -------------------- | ---
| Bug fix?             | N
| New feature?         | N
| BC breaks?           | N 
| CI currently passes? | NOPE
| Tests pass?          |  Y
| Scenarios pass?      | http://core-ci.akeneo.com/job/pim-pr-behat/2745/console
| Checkstyle issues?*  | N
| PMD issues?**        | N
| Changelog updated?   | N
| Fixed tickets        | PIM-4493
| DB schema updated?   | 
| Migration script?    |
| Doc PR               |

### CODE REVIEW

File: spec/Pim/Bundle/EnrichBundle/Provider/EmptyValue/EmptyValueProviderRegistrySpec.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        $this->addProvider($provider);
+
+        $provider->supports($attribute)->willReturn(true);
+        $provider->getEmptyValue($attribute)->willReturn('akeneo_attribute_field');
+
+        $this->getEmptyValue($attribute)->shouldReturn('akeneo_attribute_field');
+    }
+
+    function it_should_throw_an_exception_if_no_supporting_provider_is_registred(EmptyValueProviderInterface $provider, AttributeInterface $attribute)
```
Comment (skeleton) [inline spec/Pim/Bundle/EnrichBundle/Provider/EmptyValue/EmptyValueProviderRegistrySpec.php]: more than 120 characters

File: src/Pim/Bundle/EnrichBundle/Provider/EmptyValue/BaseEmptyValueProvider.php [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+class BaseEmptyValueProvider implements EmptyValueProviderInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function getEmptyValue($attribute)
+    {
+        switch ($attribute->getAttributeType()) {
```
Comment (skeleton) [inline src/Pim/Bundle/EnrichBundle/Provider/EmptyValue/BaseEmptyValueProvider.php]: I think you have forgot "pim_reference_data_multiselect"

File: src/Pim/Bundle/EnrichBundle/Normalizer/AttributeNormalizer.php [hunk 3]
```diff
@@ -24,14 +25,22 @@ class AttributeNormalizer implements NormalizerInterface
     /** @var FieldProviderInterface */
     protected $fieldProvider;
 
+    /** @var EmptyValueProviderInterface */
+    protected $emptyValueProvider;
```
Comment (jjanvier) [inline src/Pim/Bundle/EnrichBundle/Normalizer/AttributeNormalizer.php]: I guess it's a BC break (except if it's has been introduced with the PEF)
should be in the changelog

File: src/Pim/Bundle/EnrichBundle/Provider/EmptyValue/BaseEmptyValueProvider.php (hunk 2 above)
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Provider/EmptyValue/BaseEmptyValueProvider.php]: not moved away into ReferenceDataEmptyValueProvider :)

File: src/Pim/Bundle/ReferenceDataBundle/Provider/EmptyValue/ReferenceDataEmptyValueProvider.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Pim\Bundle\ReferenceDataBundle\Provider\EmptyValue;
```
Comment (jjanvier) [inline src/Pim/Bundle/ReferenceDataBundle/Provider/EmptyValue/ReferenceDataEmptyValueProvider.php]: should be in Pim\Bundle\ReferenceDataBundle\ENRICH\Provider\EmptyValue; IMO

File: src/Pim/Bundle/ReferenceDataBundle/Resources/config/providers.yml [hunk 5]
```diff
@@ -1,8 +1,14 @@
 parameters:
-    pim_reference_data.provider.field.reference_data.class: Pim\Bundle\ReferenceDataBundle\Provider\Field\ReferenceDataFieldProvider
+    pim_reference_data.provider.field.reference_data.class:       Pim\Bundle\ReferenceDataBundle\Provider\Field\ReferenceDataFieldProvider
```
Comment (jjanvier) [inline src/Pim/Bundle/ReferenceDataBundle/Resources/config/providers.yml]: .enrich.provider.... IMO

File: src/Pim/Bundle/EnrichBundle/Normalizer/AttributeNormalizer.php (hunk 3 above)
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Normalizer/AttributeNormalizer.php]: EnrichBundle/Normalizer/AttributeNormalizer.php is for the pef :)

**Pull request merged**